### ISSUE TITLE
[MOD: Cry of Fear] I can't open menu with esc button, meanwhile main menu (c_game_menu1) literally told us to press esc button for options.

### ISSUE THREAD

**Issue by TheKingFireS** [FWGS/xash3d-fwgs#1943]
self-explanatory.
In Goldsrc 4554, I can open menu with esc button while in main menu (c_game_menu1).

Info:
Patch version: 1.55 (1.2 and 1.0 untested)
Xash3D-FWGS: https://github.com/FWGS/xash3d-fwgs/commit/d6ebff81e210f6d8cd2233a4e2f0a20bb9b6abec
OS: Linux Mint 21.3 via Wine (functioned exactly like Windows 10)
GPU: GTX 1650 GDDR6
Custom renderer: Enabled
Map: c_game_menu1 (main menu)

**Comment by MohiMad** [comment]
This bug is happening inside the [Key_Event](https://github.com/FWGS/xash3d-fwgs/blob/master/engine/client/input/in_keys.c#L688) function where the engine seems to be asking the client if it wants to handle the ESC key twice:
First it asks in [line 713](https://github.com/FWGS/xash3d-fwgs/blob/master/engine/client/input/in_keys.c#L713) which the client declines (i confirmed this by adding an if statement right under which would have never have ran had the client handled the ESC key). But ESC is handled exceptionally a [bit downward in the function](https://github.com/FWGS/xash3d-fwgs/blob/master/engine/client/input/in_keys.c#L768-L788). In the case of CoF in the start menu the else-if statement `host.mouse_visible && cls.state != ca_cinematic` runs (start menu asks for a visible cursor and its not a cinematic, neither the default case or the showtexture if-statement ran. This is where it gets interesting: the engine asks the client if it wants to handle the key and returns immediately, assuming that the key is handled by the client. But the client returned 1 to the engine earlier when asked in [line 713](https://github.com/FWGS/xash3d-fwgs/blob/master/engine/client/input/in_keys.c#L713). The engine is essentially asking the same question twice and ignoring the first response. Removing the [else-if](https://github.com/FWGS/xash3d-fwgs/blob/master/engine/client/input/in_keys.c#L779-L783) statement "fixes" this issue. Hitting ESC in the main menu triggers `togglemenu`. The reason why this affects the main menu only is because its the only place where the client asks for a cursor and lets the engine handle the key (for other interfaces such as the inventory / computer interface / animal puzzle etc the client handles the key itself). 

The reason why this is a comment here and not a PR is because there's another issue that arises: when hovering over the main menu buttons (i.e. New game etc) and hitting ESC, fwgs options menu (togglemenu) opens but upon closing it the button which I had hovered over is also clicked (I did not click it). I have no idea why this happens, the single change (else-if removal) doesn't even touch the mouse-clicking events. Insights are appreciated.

**Comment by a1batross** [comment]
I don't think we need to handle ESCAPE specially in key_game case at all, so I do agree with your change, we only need it to close r_showtextures and tbh I would rather handle it as early as possible instead, so when r_showtextures is opened no input is passed to the game.

**Comment by TheKingFireS** [comment]
Fixed by #2717
